### PR TITLE
Installation Docs Restructuring

### PR DESCRIPTION
**Pull request by alexfornuto**
## Summary

- Rename quick-start directory to install
- Rename Docker document as Quickstart
-  Move Synology doc to guides

## Checklist

- [ ] reference any related issues
- [X] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [X] add appropriate tag (`improvement` / `bug` / etc)
- [X] ready for review

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -80,17 +80,16 @@ module.exports = {
           ],
         },
         {
-          title: "Quick Start",
+          title: "Install",
           collapsable: false,
-          path: "/docs/quick-start/",
+          path: "/docs/install/",
```
Comment (desimone) [inline docs/.vuepress/config.js:85]: I'd like to add redirects for the old urls to `_redirects` in `docs/.vuepress/public/_redirects`

File: docs/.vuepress/config.js [hunk 2]
```diff
@@ -165,6 +164,7 @@ module.exports = {
             "local-oidc",
             "mtls",
             "nginx",
+            "synology",
```
Comment (desimone) [inline docs/.vuepress/config.js:167]: + add redirect

**Review by desimone — CHANGES_REQUESTED** [review]

File: docs/.vuepress/config.js (hunk 2 above)
Comment (alexfornuto) [reply docs/.vuepress/config.js:167]: Good catch, added in https://github.com/pomerium/pomerium/pull/2406/commits/1a5aa5187b42110e46a5eba0edb2d9ace51da295

File: docs/.vuepress/config.js (hunk 1 above)
Comment (alexfornuto) [reply docs/.vuepress/config.js:85]: Good catch, added in https://github.com/pomerium/pomerium/pull/2406/commits/1a5aa5187b42110e46a5eba0edb2d9ace51da295

File: docs/.vuepress/public/_redirects [hunk 3]
```diff
@@ -53,3 +53,6 @@
 /jobs/Backend-Engineer.html					/careers/backend-engineer/
 
 /enterprise/								/
+
+/docs/quick-start                           /docs/install
```
Comment (desimone) [inline docs/.vuepress/public/_redirects:57]: I think this needs to be comprehensive.

Comment (alexfornuto) [reply docs/.vuepress/public/_redirects:58]: Expanded.

**Review by desimone — APPROVED** [review]

**Pull request merged**